### PR TITLE
Add patch-release JSON files

### PR DESCRIPTION
**Pull request by richlander**
I generated patch-release files for 8.0+. That was a choice. We can make a different choice.

There are links to blob storage. We can copy the files before merging or we make these GH links (it's just a code change) while we're getting this all figured out. Either works for me.

https://github.com/dotnet/core/pull/9480/files#diff-4792c8566cd38c9a12d0f0d06b0c9fc272aa7c941324a18da65dc9223b05955bR11

Contributes to https://github.com/dotnet/core/issues/9417 and borrows from https://github.com/dotnet/core/pull/9427.

@Falco20019 @leecow

### CODE REVIEW

**Comment by richlander** [comment]
We should decide what we want to do w/schemas for these new files.

Context: https://github.com/dotnet/core/issues/9431#issuecomment-2305951062

**Review by Falco20019 — APPROVED** [review]
LGTM, I'm fine with blob or GH links either way. As I'm not using them right now, I'm also fine with them only being added for 8+. Since .NET 6 is EOL soon anyway, I think that should be fine for most anyway.

**Review by admin-xspace — APPROVED** [review]

**Review by leecow — APPROVED** [review]

**Review by admin-xspace — COMMENTED** [review]
release-notes/8.0/patch-releases-index.json

**Comment by admin-xspace** [comment]
.patch

File: release-notes/9.0/patch-releases-index.json [hunk 1]
```diff
@@ -8,43 +8,43 @@
...
     },
     {
       "release-version": "9.0.0-preview.1",
       "release-date": "2024-02-13",
       "security": false,
-      "release-info-uri": "https://dotnetcli.blob.core.windows.net/dotnet/release-metadata/9.0/preview/preview1/release.json"
+      "release-info-uri": "https://raw.githubusercontent.com/dotnet/core/main/release-notes/9.0/preview/preview1/release.json"
     }
   ]
 }
```
Comment (admin-xspace) [inline release-notes/9.0/patch-releases-index.json:50]: release-notes/8.0/patch-releases-index.json

**Review by admin-xspace — COMMENTED** [review]
release-notes/8.0/patch-releases-index.json

**Comment by admin-xspace** [comment]
.patch

**Comment by richlander** [comment]
Please stop posting these messages.

**Pull request merged**